samcli/commands/deploy/guided_context.py

```diff
@@ -285,52 +288,210 @@ def prompt_parameters(
+        image_repositories: Optional[Dict[str, str]],
+        region: str,
+        s3_bucket: str,
+        s3_prefix: str,
+    ) -> Dict[str, str]:
         """
         Prompt for the image repository to push the images.
         For each image function found in build artifacts, it will prompt for an image repository.
 
         Parameters
         ----------
+        stack_name : List[Stack]
+            Name of the stack to be deployed.
+
         stacks : List[Stack]
             List of stacks to look for image functions.
 
+        image_repositories: Dict[str, str]
+            Dictionary with function logical ID as key and image repo URI as value.
+
+        region: str
+            Region for the image repos.
+
+        s3_bucket: str
+            s3 bucket URI to be used for uploading companion stack template
+
+        s3_prefix: str
+            s3 prefix to be used for uploading companion stack template
+
         Returns
         -------
-        Dict
+        Dict[str, str]
             A dictionary contains image function logical ID as key, image repository as value.
         """
-        image_repositories = {}
-        artifacts_format = get_template_artifacts_format(template_file=self.template_file)
-        if IMAGE in artifacts_format:
-            self.function_provider = SamFunctionProvider(stacks, ignore_code_extraction_warnings=True)
-            function_resources = get_template_function_resource_ids(template_file=self.template_file, artifact=IMAGE)
-            for resource_id in function_resources:
-                image_repositories[resource_id] = prompt(
-                    f"\t{self.start_bold}Image Repository for {resource_id}{self.end_bold}",
-                    default=self.image_repositories.get(resource_id, "")
-                    if isinstance(self.image_repositories, dict)
-                    else "" or self.image_repository,
-                )
-                if not is_ecr_url(image_repositories.get(resource_id)):
-                    raise GuidedDeployFailedError(
-                        f"Invalid Image Repository ECR URI: {image_repositories.get(resource_id)}"
-                    )
-            for resource_id, function_prop in self.function_provider.functions.items():
-                if function_prop.packagetype == IMAGE:
-                    image = function_prop.imageuri
-                    try:
-                        tag = tag_translation(image)
-                    except NonLocalImageException:
-                        pass
-                    except NoImageFoundException as ex:
-                        raise GuidedDeployFailedError("No images found to deploy, try running sam build") from ex
-                    else:
-                        click.secho(f"\t  {image} to be pushed to {image_repositories.get(resource_id)}:{tag}")
-            click.secho(nl=True)
+        image_repositories = image_repositories.copy() if image_repositories is not None else {}
```

Comment:
Overwriting a parameter value is usually not a good idea.
You could maybe rename the local variable to `updated_repositories` or something like that.

Reply:
Renamed to `updated_repositories`.